src/Symfony/Component/Cache/Traits/MemcachedTrait.php

```diff
@@ -49,6 +50,7 @@ private function init(\Memcached $client, $namespace, $defaultLifetime)
             }
             $this->maxIdLength -= strlen($client->getOption(\Memcached::OPT_PREFIX_KEY));
             $this->client = $client;
+            $this->isTextProtocol = !$client->getOption(\Memcached::OPT_BINARY_PROTOCOL);
```

Comment:
What do you think of always encoding the key, binary or not? That'd make the stored data independent from the actual protocol.

Reply:
It adds some overhead. But it should not be high. There two possible problems: 
1. client loses control over cache keys (mcrouter uses key names for app-level routing)
2. non-symfony and/or non-php clients would have difficulties to share cache.
I don't share memcache with non-symfony apps, so I`m ok with it.

Reply:
I think that's ok: the overhead should be negligible vs the network round-trips, and url encoding preserves most of the chars in the keys, making them still handleable.
Note that keys should be decoded on retrieval.